src/controller/CHIPDeviceController.cpp

```diff
@@ -1719,6 +1719,7 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
 #endif
         if (status != CHIP_NO_ERROR)
         {
+            actualCountryCodeSize = 2;
```

Comment:
Doesn't this also need to put "XX" back in the buffer?  Nothing guarantees buffer is unmodified by the failing call...